packages/babel-plugin-material-ui/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import * as babel from '@babel/core';
+import * as t from '@babel/types';
+import resolveImports, { MappedImportsType, ImportType } from './resolveImports';
+import { Packages } from './packages';
+
+/**
+ * Holds the import name and location of everything found in the index file of each package
+ */
+const importMaps: { [K: string]: MappedImportsType | undefined } = {};
```

Comment:
This should be instantiated in the factory so that you have one instance per babel instance and not one per module.

Reply:
This would cause it to read and process the same file over and over again though, are you sure?

Reply:
Maybe I misunderstood this plugin: Is this a stateful plugin that needs information about all files in the project?

Reply:
> Is this a stateful plugin that needs information about all files in the project?

No it isn't, the first time it comes across a `@material-ui/core` import it will parse `@material-ui/core/esm/index.js` to figure out what it exports and where it's exported from. The parsed result is stored in `importMaps`, which doesn't change from file to file